### DISCUSSION TITLE
Using local .otf file as default font

### DISCUSSION THREAD

**Discussion by Malpl3naInk** [AvaloniaUI/Avalonia#11738] [Q&A] [answered]
> I'm not a native english speaker so maybe some grammatical errors
---
I'm publishing a linux-x64 release from default template, when I running it on Ubuntu(ubntu-23.04-desktop-amd64), I got a exception called 
 `System.InvalidOperationException: Default font family name can't be null or empty` so I try to find someway to set default font

I used the method from https://github.com/AvaloniaUI/Avalonia/issues/4427#issuecomment-769767881 , It can running but still can't display Chinese

I also trying to add `.with(new FontManagerOptions(){ ... })` in BuildAvaloniaApp(), my project seems like:
```
├─Assets
│  ├─Fonts
│  │  └─NotoSansSC-Black.otf
│  │  └─...
```
So I add this (https://github.com/AvaloniaUI/Avalonia/issues/11031#issuecomment-1510915273):
```
.With(new FontManagerOptions()
{
    FontFallbacks = new []
    {
        new FontFallback
        {
            FontFamily = new FontFamily("avares://MyProject/Assets/Fonts/NotoSansSC-Black.otf#NotoSansSC-Black")
        }
    }
})
```
It throw an exception `System.UriFormatException:“Invalid URI: The format of the URI could not be determined.”
`
If I change it to `FontFamily = new FontFamily("avares://MyProject/Assets/Fonts/NotoSansSC-Black.otf")`, it can be build but still can't display

And the weird thing is, when I deleted all the changes from myself, the application still can run in ubuntu without exception, but the Chinese character changed from  ⃞ to another type
![image](https://github.com/AvaloniaUI/Avalonia/assets/80235192/d0e210c5-e8c8-451d-9eb5-50450f741a04)

So how can I using .otf file as default font and display Chinese?

**Comment by timunie** [comment]
@CrystalMoling which Avalonia version are you using?

**Reply by Malpl3naInk** [reply]
The latest stable version 0.10.21 on NuGet

**Reply by Gillibald** [reply]
Only Avalonia 11 supports this scenario

**Reply by Malpl3naInk** [reply]
So for now the only solution is using a font in system right? Or if i switch to Avalonia 11, how can I do this?

**Reply by timunie** [reply]
You can use 11.0-RC1. In your nuget manager enable the option "preview Packages" (or similar). Make sure to read the migration guide here:  https://github.com/AvaloniaUI/Avalonia/wiki/Avalonia-11-Porting-Guide

**Reply by Malpl3naInk** [reply]
I' m creating a empty project to test the Avalonia11, is this right in .csproj?
```
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <OutputType>WinExe</OutputType>
    <TargetFramework>net6.0</TargetFramework>
    <Nullable>enable</Nullable>
    <BuiltInComInteropSupport>true</BuiltInComInteropSupport>
    <ApplicationManifest>app.manifest</ApplicationManifest>
  </PropertyGroup>

  <ItemGroup>
    <ProjectCapability Include="Avalonia" />
    <TrimmerRootAssembly Include="Avalonia.Themes.Fluent" />
  </ItemGroup>

  <ItemGroup>
    <PackageReference Include="Avalonia" Version="11.0.0-rc1.1" />
    <PackageReference Include="Avalonia.Desktop" Version="0.10.21" />
    <!--Condition below is needed to remove Avalonia.Diagnostics package from build output in Release configuration.-->
    <PackageReference Condition="'$(Configuration)' == 'Debug'" Include="Avalonia.Diagnostics" Version="0.10.21" />
    <PackageReference Include="Avalonia.Fonts.Inter" Version="11.0.0-rc1.1" />
    <PackageReference Include="Avalonia.Themes.Fluent" Version="11.0.0-preview2" />
  </ItemGroup>
</Project>
```
If it's right then I have an error in Program.cs about `UsePlatformDetect()` in `public static AppBuilder BuildAvaloniaApp()` , and `BuildAvaloniaApp().StartWithClassicDesktopLifetime(args);` throws an exception of `System.InvalidOperationException:“No windowing system configured.”`

**Reply by timunie** [reply]
```diff
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <OutputType>WinExe</OutputType>
    <TargetFramework>net6.0</TargetFramework>
    <Nullable>enable</Nullable>
    <BuiltInComInteropSupport>true</BuiltInComInteropSupport>
    <ApplicationManifest>app.manifest</ApplicationManifest>
  </PropertyGroup>

  <ItemGroup>
    <ProjectCapability Include="Avalonia" />
    <TrimmerRootAssembly Include="Avalonia.Themes.Fluent" />
  </ItemGroup>

  <ItemGroup>
    <PackageReference Include="Avalonia" Version="11.0.0-rc1.1" />
    <PackageReference Include="Avalonia.Desktop" Version="0.10.21" />
    <!--Condition below is needed to remove Avalonia.Diagnostics package from build output in Release configuration.-->
    <PackageReference Condition="'$(Configuration)' == 'Debug'" Include="Avalonia.Diagnostics" Version="0.10.21" />
    <PackageReference Include="Avalonia.Fonts.Inter" Version="11.0.0-rc1.1" />
!   <PackageReference Include="Avalonia.Themes.Fluent" Version="11.0.0-preview2" />
  </ItemGroup>
</Project>
```

**Reply by Malpl3naInk** [reply]
Is this line unnecessary or I used a wrong version?

**Reply by timunie** [reply]
wrong. It references preview2 while all other packages use 11.0.0-rc1.1

**Reply by Malpl3naInk** [reply]
Oh it's my fault, I just find i searched the preview2 version instead of rc1.1 version

**Reply by Malpl3naInk** [reply]
I got an `System.IO.FileNotFoundException:“Could not load file or assembly 'Avalonia.Visuals, Version=0.10.21.0, Culture=neutral, PublicKeyToken=c8d484a7012f9a8b'. ` at `this.AttachDevTools();`
I'm looking https://github.com/AvaloniaUI/Avalonia/wiki/Avalonia-11-Porting-Guide#views but I didn't find the place to make the changes

**Reply by Gillibald** [reply]
You can change every PackageReference to use `11.0.0-rc1.1`

**Reply by timunie** [reply]
Didn't look carefully. ALL packages must have the same version. You still reference older packages.

**Reply by Malpl3naInk** [reply]
It worked but i got a transparent window
![image](https://github.com/AvaloniaUI/Avalonia/assets/80235192/3adbb777-83e3-4fc4-a419-5777955a3c13)
I found a similar issue https://github.com/AvaloniaUI/Avalonia/issues/11614
So is the correct App.axaml like this?
```
<Application xmlns="https://github.com/avaloniaui"
             xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
             x:Class="Avalonia11.App">

	<Application.Styles>
		<FluentTheme Mode="Light"/>
    </Application.Styles>
</Application>
```

**Reply by timunie** [reply, marked as answer]
Please read the poritng guide very carefully. I cannot answer every single pitfall. `<FluentTheme />` doesn't have a Mode anymore. 

Ah sorry, I should have added a link to breaking changes as well: https://github.com/AvaloniaUI/Avalonia/wiki/Breaking-Changes#avaloniathemesfluent-and-avaloniathemessimple-themes
